client/components/sites-dropdown/index.jsx

```diff
@@ -0,0 +1,89 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import classNames from 'classnames';
+import noop from 'lodash/utility/noop';
+
+/**
+ * Internal dependencies
+ */
+import Site from 'my-sites/site';
+import SiteSelector from 'components/site-selector';
+import sitesList from 'lib/sites-list';
+import Gridicon from 'components/gridicon';
+
+const sites = sitesList();
+
+export default React.createClass( {
+
+	displayName: 'SitesDropdown',
+
+	mixins: [ React.addons.PureRenderMixin ],
+
+	propTypes: {
+		selected: React.PropTypes.oneOfType( [
+			React.PropTypes.number,
+			React.PropTypes.string
+		] ),
+		showAllSites: React.PropTypes.bool,
+		indicator: React.PropTypes.bool,
+		autoFocus: React.PropTypes.bool,
+		onClose: React.PropTypes.func
+	},
+
+	getDefaultProps() {
+		return {
+			showAllSites: false,
+			indicator: false,
+			onClose: noop
+		};
+	},
+
+	getInitialState() {
+		return {
+			search: '',
+			selected: this.props.selected || sites.getPrimary().slug
+		};
+	},
+
+	getSelectedSite() {
+		return sites.getSite( this.state.selected ) || sites.getPrimary();
```

Comment:
Do we need the `|| sites.getPrimary()` part? Since `getInitialState()` sets `this.state.selected` and falls back to `sites.getPrimary().slug`, we should be fine without, no?

(This is going to be more relevant once we add filtering.)